client/client_test.go

```diff
@@ -0,0 +1,256 @@
+		t.Fatalf("no error returned when renaming non-existing file")
+	}
+}
+
+// TestDeleteFile tests the `DeleteFile` function.  Checks the indexes are
+// properly deleted and errors returned when necessary.
+func TestDeletFile(t *testing.T) {
+	client, dir := startTestClient(t)
+	defer os.RemoveAll(dir)
+
+	content := "a random content"
+	if err := ioutil.WriteFile(filepath.Join(dir, "testDeleteFile"), []byte(content), 0666); err != nil {
+		t.Fatalf("error when writing test file: %s", err)
+	}
+
+	if err := client.AddFile(filepath.Join(dir, "testDeleteFile")); err != nil {
+		t.Fatalf("error when adding the file: %s", err)
+	}
+
+	if err := client.DeleteFile(filepath.Join(dir, "testDeleteFile")); err != nil {
+		t.Fatalf("error when deleting file: %s", err)
+	}
+
+	// Doing the deleting second time should fail, as the file no longer exists
+	if err := client.DeleteFile(filepath.Join(dir, "testDeleteFile")); err == nil {
+		t.Fatalf("no error returned when deleting non-existing file")
+	}
+}
+
+// TestSearchWord tests the `SearchWord` function.  Checks that the correct set
+// of filenames are returned.
+func TestSearchWord(t *testing.T) {
+	client, dir := startTestClient(t)
+	defer os.RemoveAll(dir)
+
+	contents := []string{
+		"This is a simple test file",
+		"This is another test file",
+		"This is a different test file",
+		"This is yet another test file",
+		"This is the last test file",
+	}
+	filenames := make([]string, 5)
+
+	for i := 0; i < len(contents); i++ {
+		filenames[i] = filepath.Join(dir, "testSearchFile"+strconv.Itoa(i))
+		if err := ioutil.WriteFile(filenames[i], []byte(contents[i]), 0666); err != nil {
+			t.Fatalf("error when writing test file: %s", err)
+		}
+		if err := client.AddFile(filenames[i]); err != nil {
+			t.Fatalf("error when adding the file: %s", err)
+		}
+	}
+
+	expected := []string{filenames[1], filenames[3]}
+	sort.Strings(expected)
+	actual, err := client.SearchWord("another")
+	if err != nil {
+		t.Fatalf("error when searching word: %s", err)
+	}
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatalf("incorrect search result: expected \"%s\" actual \"%s\"", expected, actual)
+	}
+
+	empty, err := client.SearchWord("non-existing")
+	if err != nil {
+		t.Fatalf("error when searching word: %s", err)
+	}
+	if len(empty) > 0 {
+		t.Fatalf("filenames found for non-existing word")
+	}
+
+	expected = filenames
+	sort.Strings(expected)
+	actual, err = client.SearchWord("file")
+	if err != nil {
+		t.Fatalf("error when searching word: %s", err)
+	}
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatalf("incorrect search result: expected \"%s\" actual \"%s\"", expected, actual)
+	}
+}
+
+// TestMain sets up the test server and directory before the tests and tears
+// them down after the tests are completed.
+func TestMain(m *testing.M) {
+	pgid, dir, err := startTestServer()
+	if err != nil {
+		panic("error when starting the test server")
+	}
+
+	// Redirect the logs from the client to nil
```

Comment:
Why do we want this?  What in the client is logging to stderr during the tests anyway?